src/analysis/mesh/qgsmeshtriangulation.cpp

```diff
@@ -328,3 +334,107 @@ int QgsMeshZValueDataset::valuesCount() const
 {
   return mMesh.vertexCount();
 }
+
+QgsMeshEditingDelaunayTriangulation::QgsMeshEditingDelaunayTriangulation() = default;
+
+QgsTopologicalMesh::Changes QgsMeshEditingDelaunayTriangulation::apply( QgsMeshEditor *meshEditor )
+{
+  //use only vertices that are on boundary or free, if boundary
+  QList<int> vertexIndextoTriangulate;
+
+  QList<int> removedVerticesFromTriangulation;
+
+  for ( const int vertexIndex : std::as_const( mInputVertices ) )
+  {
+    if ( meshEditor->isVertexFree( vertexIndex ) || meshEditor->isVertexOnBoundary( vertexIndex ) )
+      vertexIndextoTriangulate.append( vertexIndex );
+    else
+      removedVerticesFromTriangulation.append( vertexIndex );
+  }
+
+  bool triangulationReady = false;
+  bool giveUp = false;
```

Comment:
shouldn't this be (re)set in the while loop too? or it is one-off giveUp?

Reply:
one-off